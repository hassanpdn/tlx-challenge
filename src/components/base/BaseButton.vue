<template>
      <button :class="variant" :disabled="disabled" @click="callback($event)">
            <span>{{title}}</span>
      </button>
</template>

<script>
export default {
      name: "btn",
      props: {
            variant: {
                  type: String,
                  default: 'normal'
            },
            title: {
                  type: String
            },
            disabled: {
                  type: Boolean,
                  default: false
            },
      },
      methods: {
            callback: function (e) {
                  this.$emit("click", e);
            },
      },
};
</script>

<style>
      button {
            padding: 7px 14px;
            border: 1px solid transparent;
            color: var(--darker-gray);
            border-radius: var(--radius-sm);
            cursor: pointer;
      }
      button.disabled {
            cursor: not-allowed;
      }
      button.danger {
            background-color: var(--danger);
            color: var(--white);
      }
      button.success {
            background-color: var(--success);
            color: var(--white);
      }
      button.info {
            background-color: var(--info);
            color: var(--white);
      }
      button.warning {
            background-color: var(--warning);
            color: var(--white);
      }
      button:hover {
            box-shadow: 0 3px 21px -7px var(--gray);
            transition: ease .3s;
      }
</style>
