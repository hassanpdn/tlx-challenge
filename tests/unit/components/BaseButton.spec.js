import { mount } from '@vue/test-utils';
import BaseButton from '@/components/base/BaseButton';

describe('BaseButton', () => {
      it("renders component", () => {
            const wrapper = mount(BaseButton, {propsData:{
                  variant: 'danger',
                  icon: 'angleRight',
                  title: 'test',
                  disabled: false
            }});
            expect(wrapper.element).toMatchSnapshot()
      })
      it("variant has initial variant of normal", () => {
            const wrapper = mount(BaseButton, {propsData:{
                  icon: 'angleRight',
                  title: 'test',
                  disabled: false
            }});
            expect(wrapper.vm.variant).toBe('normal')
      })
});
