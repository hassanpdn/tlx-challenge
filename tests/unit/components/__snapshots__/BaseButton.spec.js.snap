// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton renders component 1`] = `
<button
  class="btn danger"
>
  <span>
    test
  </span>
   
  <svg
    class="icon icon is-button"
    viewBox="0 0 256 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
      fill="#fff"
      opacity="1"
    />
  </svg>
</button>
`;
